src/classes/ADDR_Account_TDTM.cls

```diff
@@ -47,124 +70,183 @@ public class ADDR_Account_TDTM extends TDTM_Runnable {
-                if (UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c != null && acc.RecordTypeId != null
-                && UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c.contains(acc.RecordTypeId)) {
-                    if (isAccountAddressSpecified(acc))
-                        listAccCreateAddr.add(acc);
-                }
-            }
-    
-            // AFTER UPDATE
-            if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
-                Account accOld = (Account)listOld[i];
-                // note that we detect whether the address change is coming from the address trigger,
-                // in which case we don't want to create a new address.
-                if (ADDR_Addresses_TDTM.hasRunAddrTrigger) {
-                    continue;
-                }
-                if (UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c != null && acc.RecordTypeId != null
-                && UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c.contains(acc.RecordTypeId)) {
-                    // if the address changed, remember the account we want to add a new address for    
-                    if (isAccountAddressChanged(acc, accOld)) 
-                        listAccCreateAddr.add(acc);
-                }
-            }                
+        if(!alreadyRunAfterInsert || !alreadyRunAfterUpdate) {
+	        
+	        //Turn off other address triggers
+            ADDR_Contact_TDTM.turnOff();
+            ADDR_Addresses_TDTM.turnOff();
+        
+	        List<Account> newAddrInfoAccs = new List<Account>();      
+	        List<Account> accsAddrInfoCleared = new List<Account>();
+	                
+	        // AFTER INSERT
+	        if (!alreadyRunAfterInsert && triggerAction == TDTM_Runnable.Action.AfterInsert) {
+	            Integer i = 0;        
+		        for (SObject so : listNew) {
+		            Account acc = (Account)so;
+	                // for new accounts that are using address management, create the address object.
+	                if (UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c != null && acc.RecordTypeId != null
+	                && UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c.contains(acc.RecordTypeId)) {
+	                    if (!ADDR_Addresses_UTIL.isAccAddressEmpty(acc)) {
+	                        newAddrInfoAccs.add(acc);
+	                    }
+	                }
+	                i++;
+		        }
+	            alreadyRunAfterInsert = true;
+	        }
+		    
+	        // AFTER UPDATE
+	        if (!alreadyRunAfterUpdate && triggerAction == TDTM_Runnable.Action.AfterUpdate) {  
+	            Integer i = 0;        
+	            for (SObject so : listNew) {
+	                Account acc = (Account)so;
+	                Account accOld = (Account)listOld[i];
+	                Boolean isHousehold = acc.RecordTypeId == UTIL_Describe.getHhAccRecTypeID();
```

Comment:
does HEDA support not having record types for HH Account, and just using the system Account Type field?  if so, then this test would need to change.